app/services/commitments/helper_service.rb

```diff
@@ -0,0 +1,79 @@
+# frozen_string_literal: true
+
+module Commitments
+  class HelperService < BaseService
+    def initialize(commitment:, invoice_subscription:, current_usage: false)
+      @commitment = commitment
+      @invoice_subscription = invoice_subscription
+
+      super
+    end
+
+    def proration_coefficient
+      result.proration_coefficient = calculate_proration_coefficient
+      result
+    end
+
+    def period_invoice_ids
+      result.period_invoice_ids = get_period_invoice_ids
+      result
+    end
+
+    private
+
+    attr_reader :commitment, :invoice_subscription, :current_usage
+
+    delegate :subscription, to: :invoice_subscription
+
+    def calculate_proration_coefficient
+      all_invoice_subscriptions = subscription
+        .invoice_subscriptions
+        .where(invoice_id: get_period_invoice_ids)
+        .order(
+          Arel.sql(
+            ActiveRecord::Base.sanitize_sql_for_conditions(
+              'COALESCE(invoice_subscriptions.to_datetime, invoice_subscriptions.created_at) ASC',
+            )
+          )
+        )
+
+      days = Utils::DatetimeService.date_diff_with_timezone(
+        all_invoice_subscriptions.first.from_datetime,
+        invoice_subscription.to_datetime,
+        subscription.customer.applicable_timezone,
+      )
+
+      days_total = Utils::DatetimeService.date_diff_with_timezone(
+        dates_service.previous_beginning_of_period,
+        dates_service.end_of_period,
+        subscription.customer.applicable_timezone,
+      )
+
+      days / days_total.to_f
+    end
+
+    def get_period_invoice_ids
```

Comment:
Naming/AccessorMethodName: Do not prefix reader method names with `get_`. (https://rubystyle.guide#accessor_mutator_method_names)